common-protobuf/src/main/java/io/protostuff/runtime/RuntimeMapFieldProtobuf.java

```diff
@@ -0,0 +1,166 @@
+import io.protostuff.Input;
+import io.protostuff.MapSchema;
+import io.protostuff.MapSchema.MapWrapper;
+import io.protostuff.MapSchemaUtils;
+import io.protostuff.Output;
+import io.protostuff.Pipe;
+import io.protostuff.ProtostuffException;
+import io.protostuff.Schema;
+
+public class RuntimeMapFieldProtobuf<T> extends RuntimeMapField<T, Object, Object> {
+    private static java.lang.reflect.Field entrySchemaField =
+        ReflectionUtils.findField(MapSchema.class, "entrySchema");
+    static {
+        entrySchemaField.setAccessible(true);
+    }
+
+    private RuntimeMapField<T, Object, Object> runtimeMapField;
+
+    private java.lang.reflect.Field field;
+
+    private Schema<Entry<Object, Object>> entrySchema;
+
+    @SuppressWarnings("unchecked")
+    public RuntimeMapFieldProtobuf(RuntimeMapField<T, Object, Object> runtimeMapField,
+            java.lang.reflect.Field field) {
+        super(runtimeMapField.type, runtimeMapField.number, runtimeMapField.name, null,
+                runtimeMapField.schema.messageFactory);
+
+        entrySchema = (Schema<Entry<Object, Object>>) ReflectionUtils.getField(entrySchemaField,
+                runtimeMapField.schema);
+
+        this.runtimeMapField = runtimeMapField;
+        this.field = field;
+    }
+
+    @Override
+    protected void mergeFrom(Input input, T message) throws IOException {
+        if (!ProtobufFeatureUtils.isUseProtobufMapCodec()) {
+            runtimeMapField.mergeFrom(input, message);
+            return;
+        }
+
+        try {
+            @SuppressWarnings("unchecked")
+            Map<Object, Object> value = (Map<Object, Object>) field.get(message);
+            if (value == null) {
+                value = schema.newMessage();
+                field.set(message, value);
+            }
+
+            MapWrapper<Object, Object> mapWrapper = MapSchemaUtils.createMapWrapper(value);
+            if (ByteArrayInput.class.isInstance(input)) {
+                ((ByteArrayInput) input).readRawVarint32();
+            } else if (ByteBufferInput.class.isInstance(input)) {
+                ((ByteBufferInput) input).readRawVarint32();
+            } else {
+                throw new Error("not handler " + input.getClass().getName());
+            }
+
+            int keyNumber = input.readFieldNumber(schema);
+            if (keyNumber != 1) {
+                throw new ProtostuffException(
+                        "The map was incorrectly serialized, expect key number 1, but be " + keyNumber);
+            }
+            Object key = kFrom(input, null);
+
+            int valueNumber = input.readFieldNumber(schema);
+            if (valueNumber != 2) {
+                throw new ProtostuffException(
+                        "The map was incorrectly serialized, expect value number 2, but be " + valueNumber);
+            }
+            vPutFrom(input, mapWrapper, key);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
only field reflect exception need to be catch.
so catch and throw ProtostuffException